lib/sample_uploader/utils/transformations.py

```diff
@@ -0,0 +1,90 @@
+from installed_clients.OntologyAPIClient import OntologyAPI
+from sample_uploader.utils.samples_content_error import SampleContentError
+import time
+import json
+import pandas as pd
+
+_ENVO_ONTOLOGY = 'envo_ontology'
+_ENVO_ID_PREFIX = 'ENVO:'
+
+_ONTOLOGY_COLS = {
+    'enigma:material': {
+        'ontology': _ENVO_ONTOLOGY,
+        'id_prefix': _ENVO_ID_PREFIX
+    },
+    'biome': {
+        'ontology': _ENVO_ONTOLOGY,
+        'id_prefix': _ENVO_ID_PREFIX
+    },
+    'feature': {
+        'ontology': _ENVO_ONTOLOGY,
+        'id_prefix': _ENVO_ID_PREFIX
+    }
+}
+
+
+def _get_timestamp():
+    return int(time.time() * 1000)
+
+
+class FieldTransformer:
+    def __init__(self, callback_url):
+        '''
+        '''
+        self.onto_api = OntologyAPI(callback_url)
+
+    def _ontology_field_transforms(self, row, cols):
+        '''
+        Transformations related to fields that are validated against an ontology.
+        params:
+            row - pd.Series, 1 row of a pd.DataFrame.
+            cols - all metadata columns of the input DataFrame/Row
+        '''
+        # find which ontology_cols are in the row.
+        onto_cols = set(cols).intersection(set(_ONTOLOGY_COLS.keys()))
+        for key in onto_cols:
+            # first check if already in 'id' form.
+            if not row.get(key) or pd.isnull(row[key]):
+                continue
+            onto_val = str(row[key])
+            values = _ONTOLOGY_COLS.get(key)
+            query_ontology = values.get('ontology')
+            id_prefix = values.get('id_prefix')
+            if onto_val.startswith(id_prefix):
+                continue
+            # check if in appropriate ontology.
+            params = {
+                'name': onto_val,
+                # 'ancestor_term': ,
+                'ts': _get_timestamp(),
+                'ns': query_ontology
+            }
+            ret = self.onto_api.get_term_by_name(params)
+            # ensure there is only 1 result
+            if len(ret['results']) != 1:
+                raise RuntimeError(f"Received {len(ret['results'])} results from query with params {params} in OntologyAPI, Expected 1 result.")
```

Comment:
The following should be enough to make things work (Also might be worth putting a bit clearer of an err message for the user, I did my best there):
```suggestion
                raise SampleContentError(f"Couldn't resolve ontology term. Received {len(ret['results'])} results from query with params {params} in OntologyAPI, Expected 1 result.", key=key)
```